Code/GraphMol/RGroupDecomposition/RGroupCore.cpp

```diff
@@ -418,6 +418,22 @@ void RCore::buildMatchingMol() {
     // then keep one attachment
     if (atom->getAtomicNum() == 0 && atom->getDegree() == 1 &&
         isDummyRGroupAttachment(*atom)) {
+      // if we are a query in disguise, don't strip
+      if(atom->hasQuery()) {
+        auto q = atom->getQuery()->getDescription();
+	// If we are anything but AtomNull or AtomAtomicNum we are not
+	//  fully a dummy atom
+	if(q != "AtomNull" && q != "AtomAtomicNum") {
+	  unsigned int type=0xFFFFFFFF;
+	  //  If we are were set as an RLABEL of type Isotope we probably
+	  //   have a query of named "AtomIsotope"
+	  //   we DO strip these unless they have an additional query
+	  if(!atom->getPropIfPresent(RLABEL_TYPE, type) || type != RGroupLabels::IsotopeLabels || q != "AtomIsotope") {
```

Comment:
I generally prefer to do the simple checks first in the if:
```suggestion
	  if(q != "AtomIsotope" || !atom->getPropIfPresent(RLABEL_TYPE, type) || type != RGroupLabels::IsotopeLabels) {
```